packages/@ourworldindata/components/tsconfig.json

```diff
@@ -1,6 +1,7 @@
 {
     "extends": "../../../devTools/tsconfigs/tsconfig.base.json",
     "compilerOptions": {
+        "module": "es2022",
```

Comment:
Why did we go for this option? The [TS docs](https://www.typescriptlang.org/tsconfig/#module) say:

> You very likely want `nodenext` for modern Node.js projects and `preserve` or `esnext` for code that will be bundled.

We already use `NodeNext` in `functions/tsconfig.json`.

And is it not possible to change it everywhere by modifying the `devTools/tsconfigs/tsconfig.base.json` instead? Is there a reason to keep `commonjs` there? If it's because Vite uses that value, could we override it to `commonjs` in the `site` only?